### PR TITLE
Enhance calendar

### LINKED ISSUE #355: Title: Enhancement in the project CALENDER.
### Is your proposal related to a problem? Please describe.

I would like to work in the quality of the code of project **CALENDER**. https://github.com/Kushal997-das/Project-Guidance/tree/main/Desktop%20Application/Basic/C%2B%2B/Calender

### Add any other context or screenshots about the proposal request here.

_No response_

### PR DESCRIPTION
**Pull request by Satyam26-09**
### Have you read the [Contributing Guidelines](https://github.com/Kushal997-das/Project-Guidance/blob/main/CONTRIBUTING.md) ?

YES

### Description

- program will end when user input is negative in year as the program not work for negative years.

- restrict the program till 1000000000 years. 

- program will re-prompt for month again if user input is not between 1-12.

Fixes #355

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue: https://github.com/Kushal997-das/Project-Guidance/issues/355

PR made under GSSoC'22 contributions

### CODE REVIEW

File: Desktop Application/Basic/C++/Calender/Calender.cpp [hunk 1]
```diff
@@ -4,11 +4,18 @@ int main()
...
+        return 1;
+    }
     //Month for which the calender is required
-    std::cout << "Enter the number of the month ";
-    std::cin >> a;
+    do
+    {
+        std::cout << "Enter the number of the month :";
+        std::cin >> a;
+    } while (a<1 || a>12);
```
Comment (Kushal997-das) [inline Desktop Application/Basic/C++/Calender/Calender.cpp:18]: Still there is an error 
- when you run the program it's asking for `"Enter the year (from 0 to 1000000000) :` but when user enter any negative number it just terminate without saying anything fix this one .
- Then it asking for `Enter the number of the month` but if user enter beyond limit the program still continue like you have given limit `1 to 12` but what if user enter 0 or more than 12 or maybe any negative number , so fix this also write an edge case for it .

File: Desktop Application/Basic/C++/Calender/README.md [hunk 2]
```diff
@@ -4,6 +4,7 @@ Developer: Ayushi
 
 If you are getting stuck becuase you can't find the calender nearby you.
 Then this is the right application you must have.
-Just enter the month and year number and your caleder is ready.
+Just enter the month and year number (from 0 to 1000000000) and your caleder is ready.
 
 <img width="393" alt="Screenshot 2022-03-02 at 1 47 41 PM" src="https://user-images.githubusercontent.com/93306058/156322283-d91f6d37-baf0-4c7e-8142-eb7ca7f2fb81.png">
+![Screenshot 2022-03-13 135439](https://user-images.githubusercontent.com/96687053/158051989-3d9bbbc3-73d6-408d-8aae-2fb02867e816.png)
```
Comment (Kushal997-das) [inline Desktop Application/Basic/C++/Calender/README.md:10]: Delete this one add newly updated one after fixing bugs.

**Review by Kushal997-das — CHANGES_REQUESTED** [review]
@Satyam26-09 do the changes !!

File: Desktop Application/Basic/C++/Calender/Calender.cpp (hunk 1 above)
Comment (Satyam26-09) [reply Desktop Application/Basic/C++/Calender/Calender.cpp:19]: Done...

File: Desktop Application/Basic/C++/Calender/README.md (hunk 2 above)
Comment (Satyam26-09) [reply Desktop Application/Basic/C++/Calender/README.md]: Done..

File: Desktop Application/Basic/C++/Calender/Calender.cpp [hunk 3]
```diff
@@ -4,11 +4,19 @@ int main()
 {
     int a, c, s, i, x, b, l, r, y, m, p;
     //Year 
-    std::cout << "Enter the year ";
+    std::cout << "Enter the year (from 0 to 1000000000) :";
     std::cin >> b;
+    if (b<0 || b>1000000000)
```
Comment (Kushal997-das) [inline Desktop Application/Basic/C++/Calender/Calender.cpp:9]: How year can be 0 ? Your range should start with valid year .

File: Desktop Application/Basic/C++/Calender/Calender.cpp [hunk 4]
```diff
@@ -4,11 +4,19 @@ int main()
...
+    {
+        std::cout<<"ERROR::The input year is not in range range.\n"<<"EXIT";
+        return 1;
+    }
     //Month for which the calender is required
-    std::cout << "Enter the number of the month ";
-    std::cin >> a;
+    do
+    {
+        std::cout << "Enter the number of the month(from 1 to 12) :";
```
Comment (Kushal997-das) [inline Desktop Application/Basic/C++/Calender/Calender.cpp:17]: AS i mentioned early your month have still same problem .

File: Desktop Application/Basic/C++/Calender/Calender.cpp (hunk 3 above)
Comment (Satyam26-09) [reply Desktop Application/Basic/C++/Calender/Calender.cpp:9]: If we consider BC era then negative year is still possible. But calendar shown for that is not valid. A valid calendar starts from 0 year.
Do you still want me to change it?

Comment (Satyam26-09) [reply Desktop Application/Basic/C++/Calender/Calender.cpp:9]: @Kushal997-das

File: Desktop Application/Basic/C++/Calender/Calender.cpp (hunk 4 above)
Comment (Satyam26-09) [reply Desktop Application/Basic/C++/Calender/Calender.cpp:17]: do you want me to end the program if the input month is invalid?

Comment (Kushal997-das) [reply Desktop Application/Basic/C++/Calender/Calender.cpp:17]: > do you want me to end the program if the input month is invalid?

Yes , with a message .

File: Desktop Application/Basic/C++/Calender/README.md [hunk 5]
```diff
@@ -4,6 +4,8 @@ Developer: Ayushi
 
 If you are getting stuck becuase you can't find the calender nearby you.
 Then this is the right application you must have.
-Just enter the month and year number and your caleder is ready.
+Just enter the month and year number (from 0 to 1000000000) and your caleder is ready.
 
 <img width="393" alt="Screenshot 2022-03-02 at 1 47 41 PM" src="https://user-images.githubusercontent.com/93306058/156322283-d91f6d37-baf0-4c7e-8142-eb7ca7f2fb81.png">
```
Comment (Kushal997-das) [inline Desktop Application/Basic/C++/Calender/README.md:9]: Delete this!

File: Desktop Application/Basic/C++/Calender/README.md [hunk 6]
```diff
@@ -4,6 +4,8 @@ Developer: Ayushi
...
+
+![Screenshot 2022-03-14 092955](https://user-images.githubusercontent.com/96687053/158102734-acacf7a1-52b4-48b8-86f4-ce6d74757047.png)
```
Comment (Kushal997-das) [inline Desktop Application/Basic/C++/Calender/README.md:11]: Delete this one as well , and update with new one .

**Review by Kushal997-das — CHANGES_REQUESTED** [review]
@Satyam26-09 Just update the pictures with updated the program everything else looks correct .

**Comment by Satyam26-09** [comment]
@Kushal997-das  done...kindly check.

**Review by Kushal997-das — APPROVED** [review]
@Satyam26-09 Great !

**Comment by Satyam26-09** [comment]
@Kushal997-das thank you :)

**Pull request merged**